torch/ao/pruning/_experimental/pruner/README.md

```diff
@@ -1,93 +1,263 @@
-# Intro
+# Structured Pruning
+
+## Intro / Motivation
+
+**pruning** is the process of removing weights / connections in a neural network in order to make it run faster.
+
+There are two types of pruning that we support: **structured pruning** and **unstructured pruing**
+
+* **Unstructured pruning** refers to pruning any arbritary weight
+* **Structured pruning** refers to pruning by removing entire rows / columns of a weight.
+
+Structured pruning offers several advantages over unstructured pruning at the cost of lower granuality:
+1. We can store the masks more efficeintly in structured pruning.
+2. Structured pruning relies on removing columns / rows to a speedup whereas unstructured pruning requires custom kernels.
+
+Pruning is usually followed by a retraining or fine-tuning step.
```

Comment:
this line feels a bit out of place, perhaps walk a user through a generic pruning process??